File: rust/examples/linux/display/pcf8576/minimal/src/main.rs
use linux_embedded_hal::I2cdev;
use periph::chips::display::{Pcf8576Minimal, SEVEN_SEG};

fn main() {
    let i2c_bus: u8 = std::env::var("I2C_BUS").ok().and_then(|v| v.parse().ok()).unwrap_or(1);
    let addr: u8 = std::env::var("I2C_ADDR")
        .ok()
        .and_then(|v| u8::from_str_radix(v.trim_start_matches("0x"), 16).ok())
        .unwrap_or(0x38);

    let dev = I2cdev::new(format!("/dev/i2c-{}", i2c_bus)).expect("open i2c bus");
    let mut lcd = Pcf8576Minimal::new(dev, addr).expect("init PCF8576"); // Create PCF8576 driver, (i2c, addr=0x38)

    let digits = [1u8, 2, 3, 4];
    for (i, d) in digits.iter().enumerate() {
        let seg = SEVEN_SEG[*d as usize]; // Encode 7-segment digit, (digit 0–9) → u8
        lcd.set_digit_7seg(i as u8, seg).expect("write digit"); // Write one digit, (position 0–19, segments 0–255) → ()
    }
}

File: cpp/examples/linux/display/PCF8576/minimal/main.cpp
#include <cstdio>
#include <cstdlib>
#include <unistd.h>
#include "I2CConnectionLinux.h"
#include "PCF8576.h"

int main() {
    const char* bus_env  = getenv("I2C_BUS");
    const char* addr_env = getenv("I2C_ADDR");
    int     bus  = bus_env  ? atoi(bus_env)       : 1;
    uint8_t addr = addr_env ? (uint8_t)strtol(addr_env, nullptr, 0) : 0x38;
    I2CConnectionLinux connection(bus, addr);

    PCF8576Minimal pcf(connection);                                         // Create PCF8576 driver, (connection)
                                                                           // 1:4 multiplex, 1/3 bias, display enabled

    const uint8_t digits[4] = {1, 2, 3, 4};
    for (uint8_t pos = 0; pos < 4; pos++)
        pcf.set_digit_7seg(pos, PCF8576Minimal::SEVEN_SEG[digits[pos]]);   // Write one 7-segment digit, (position, segments) → void
    usleep(2000000);
    pcf.clear();                                                           // Clear display RAM, () → void
    return 0;
}
